### PR TITLE
Added Docker to Database Getting Started

### PR DESCRIPTION
**Pull request by awanlin**
## Hey, I just made a Pull Request!

This adds alternative instructions for running Postgres in a Docker container, this is a really handy and flexible option.

Also added a link to install pgAdmin 4 also super handy!

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/getting-started/config/database.md [hunk 1]
```diff
@@ -68,17 +68,27 @@ That's enough database administration to get started. Type `\q`, followed by
 pressing the enter key. Then again type `exit` and press enter. Next, you need
 to install and configure the client.
 
-### 2. Configuring Backstage `pg` Client
+### Docker Alternative
```
Comment (Parsifal-M) [inline docs/getting-started/config/database.md:71]: This is awesome and I think really useful, I was wondering how we feel about also suggesting a `docker compose` file? We use it and its so easy 😄  e.g.

```yaml
version: '4'
services:
  postgres:
    image: postgres:15.5-alpine
    environment:
      POSTGRES_USER: localpostgres
      POSTGRES_PASSWORD: **********
    ports:
      - 5432:5432
 ```

Then you just `up` or `down` when ever you need it.

But this is also fine, I think either way is super helpful 👏

**Review by Parsifal-M — COMMENTED** [review]
Awesome! 👏  just a couple of questions 💭

File: docs/getting-started/config/database.md [hunk 2]
```diff
@@ -68,17 +68,27 @@ That's enough database administration to get started. Type `\q`, followed by
...
 
-Go to the root directory of your freshly installed Backstage
-App. Run the following to install the PostgreSQL client into your backend:
+Alternatively you can run Postgres in a Docker container, this is great for local development or getting a Backstage POC up and running quickly, here's how:
 
-```bash title="From your Backstage root directory"
-yarn --cwd packages/backend add pg
+First we need to pull down the container image, we'll use Postgres 17, check out the [Postgres Version Policy](../../overview/versioning-policy.md#postgresql-releases) to learn which versions as supported.
```
Comment (ambar-writer) [inline docs/getting-started/config/database.md:75]: s/versions as supported/versions are supported/

**Review by ambar-writer — COMMENTED** [review]
One more bit of fluff to pick off --

**Review by freben — APPROVED** [review]
Approving on my end, good other comments here!

**Comment by freben** [comment]
@awanlin ping on this one, some small things to tweak i think?

File: docs/getting-started/config/database.md [hunk 3]
```diff
@@ -23,7 +23,7 @@ This guide assumes a basic understanding of working on a Linux based operating s
 - If the database is not hosted on the same server as the Backstage app, the
   PostgreSQL port needs to be accessible (the default is `5432` or `5433`)
 
-### 1. Install and configure PostgreSQL
+## 1. Install and configure PostgreSQL
```
[the remark this answers is not in the document]
Comment (awanlin) [reply docs/getting-started/config/database.md]: Yup, good call, corrected 👍

File: docs/getting-started/config/database.md (hunk 1 above)
Comment (awanlin) [reply docs/getting-started/config/database.md]: Cool, I've adjusted the structure to accommodate this a bit more easily IMO. Can you give this another look please?

File: docs/getting-started/config/database.md (hunk 2 above)
Comment (awanlin) [reply docs/getting-started/config/database.md]: Nice, good catch, fix this up with my latest changes 👍

**Pull request merged**